protocol/error.go

```diff
@@ -36,26 +34,26 @@ var ErrorResponses = map[ErrorCode]bool{
 }
 
 var (
-	errorMessages = map[ErrorCode]error{
-		Success:                     nil,
-		ErrorMalformedClientMessage: errors.New("[coniks] Malformed client request"),
-		ErrorNameExisted:            errors.New("[coniks] Registering identity is already registered"),
-		ErrorNameNotFound:           errors.New("[coniks] Name not found"),
-		ErrorDirectory:              errors.New("[coniks] Directory error"),
-
-		Passed: nil,
-		ErrorMalformedDirectoryMessage: errors.New("[coniks] Malformed directory message"),
-		ErrorBadSignature:              errors.New("[coniks] Directory's signature is invalid"),
-		ErrorBadVRFProof:               errors.New("[coniks] Bad VRF proof"),
-		ErrorBadIndex:                  errors.New("[coniks] Directory returned a bad index"),
-		ErrorBadMapping:                errors.New("[coniks] Returned name-to-key mapping is inconsistent with the root hash"),
-		ErrorBadSTR:                    errors.New("[coniks] The hash chain is inconsistent"),
-		ErrorBadCommitment:             errors.New("[coniks] Bad commitment"),
-		ErrorBadBinding:                errors.New("[coniks] Bad name-to-key binding"),
-		ErrorCouldNotVerify:            errors.New("[coniks] Could not verify"),
+	errorMessages = map[ErrorCode]string{
+		Success:                     "[coniks] Success",
+		ErrorMalformedClientMessage: "[coniks] Malformed client request",
+		ErrorNameExisted:            "[coniks] Registering identity is already registered",
+		ErrorNameNotFound:           "[coniks] Name not found",
+		ErrorDirectory:              "[coniks] Directory error",
+
+		Passed: "[coniks] Passed",
+		ErrorMalformedDirectoryMessage: "[coniks] Malformed directory message",
+		ErrorBadSignature:              "[coniks] Directory's signature is invalid",
+		ErrorBadVRFProof:               "[coniks] Bad VRF proof",
+		ErrorBadIndex:                  "[coniks] Directory returned a bad index",
+		ErrorBadMapping:                "[coniks] Returned name-to-key mapping is inconsistent with the root hash",
+		ErrorBadSTR:                    "[coniks] The hash chain is inconsistent",
+		ErrorBadCommitment:             "[coniks] Bad commitment",
```

Comment:
Much like in my previous comments, I would make the error message for `ErrorBadCommitment` a bit more informative.